control-plane/csi-driver/src/bin/node/filesystem_ops.rs

```diff
@@ -0,0 +1,276 @@
+//! This module consists of the various filesystem specific operations utility. Including creation
+//! of filesystem, changing the parameter of filesystem like uuid, default mount options for
+//! specific filesystem, repairing of the filesystem, retrieving specific properties of the
+//! filesystem.
+
+use crate::mount;
+use csi_driver::filesystem::FileSystem as Fs;
+use devinfo::{blkid::probe::Probe, DevInfoError};
+use std::process::Output;
+use tokio::process::Command;
+use tonic::async_trait;
+use tracing::trace;
+use uuid::Uuid;
+
+/// `nouuid` mount flag, to allow duplicate fs uuid.
+const NO_UUID_FLAG: &str = "nouuid";
+
+/// Error type filesystem operations.
+type Error = String;
+
+/// Ext4 filesystem type.
+pub(crate) struct Ext4Fs;
+/// XFS filesystem type.
+pub(crate) struct XFs;
+/// Unsupported filesystem type.
+struct Unsupported;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) struct FileSystem(pub(crate) Fs);
```

Comment:
missing doc comment

make inner field private and add from_fs|from_str method instead?